x-pack/plugins/security_solution/public/timelines/components/flyout/action_menu/index.tsx

```diff
@@ -63,6 +65,16 @@ const TimelineActionMenuComponent = ({
           title=""
         />
       </EuiFlexItem>
+      {userCasesPermissions.create && userCasesPermissions.read ? (
+        <>
+          <EuiFlexItem>
+            <VerticalDivider />
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <AddToCaseButton timelineId={timelineId} />
+          </EuiFlexItem>
+        </>
+      ) : null}
```

Comment:
Few years back, I learned a lesson hard way to use ternaries generally instead of `&&` operations in react components specifically because it can sometime return values unpredictably which will result in rendering in DOM. Consider below example. What do think the output of below component should be when array is empty v/s array is filled.

```js
// for example

function ArrayFilledText(arr: Array<any>){
  return arr.length && 'Filled'
}
```

In above case, array will return, `Filled` if array.length is > 0 , else it will return 0 which will render in the DOM.

[Here is one article which ](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx)explains the issue much better than I can.

I do think that mostly it should be fine. But I follow this personal rule to **always** use ternaries when returning from a component to be safe. If you are not particularly against, I would like to keep it as it is.

Reply:
makes sense, thanks for the explanation! :)